config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/PolicyEntityBuilder.java

```diff
@@ -509,6 +510,36 @@ private void prepareAssertion(Element policyElement, String assertionTag, Consum
         }
     }
 
+    @VisibleForTesting
+    void prepareRoutingAssertionCertificateIds(Document policyDocument, Bundle bundle, Element assertionElement) {
```

Comment:
In future, we will have to implement it for each assertion? If yes, we should think of refactoring it so that we just write an implementation for each assertion like Simplifier.

Reply:
@nikhilahuja , I am not sure whether we can handle it commonly all the assertions, each assertion may represents the environment Id's differently and it may not be certificate ID's

Reply:
I am not talking about generic but we can have an interface and then we run all the implementations on this policy. similar to Simplifier or EntityLoaders or Linkers. Each assertion implementation that needs to do changes like this will just have to implement that interface.